### PR TITLE
add ../ to candidates

### PR DESCRIPTION
**Pull request by gamark**
default:
[in]
../
some directories
some files

### CODE REVIEW

**Comment by Shougo** [comment]
Sorry.  I don't want to merge the feature.
Why it is really needed?  Please describe it more clearly.

The feature exists in vimfiler.  But I have removed it in defx.

**Comment by Shougo** [comment]
It must be optional feature at least.

**Comment by gamark** [comment]
> It must be optional feature at least.

it for beginner friendly.

File: doc/defx.txt [hunk 1]
```diff
@@ -784,6 +784,10 @@ OPTIONS							*defx-options*
 		"floating".
 
 		Default: "none"
+						*defx-option-show-parent*
+-show-parent
```
Comment (Shougo) [inline doc/defx.txt:788]: Please sort the options order.

File: doc/defx.txt [hunk 2]
```diff
@@ -784,6 +784,10 @@ OPTIONS							*defx-options*
...
+		Show ../ parent directory by default.
```
Comment (Shougo) [inline doc/defx.txt:789]: `by default` is not needed.

Comment (Shougo) [reply doc/defx.txt:789]: `Show parent "../" directory in the buffer` is better for me.

File: rplugin/python3/defx/context.py [hunk 3]
```diff
@@ -52,3 +52,4 @@ class Context(typing.NamedTuple):
     winwidth: int = 0
     vertical_preview: bool = False
     winborder: str = 'none'
+    show_parent:bool = False
```
Comment (Shougo) [inline rplugin/python3/defx/context.py:55]: Please sort the name.

File: rplugin/python3/defx/source/file/__init__.py [hunk 4]
```diff
@@ -60,6 +60,8 @@ def gather_candidates(
                     'is_directory': safe_call(entry.is_dir, False),
                     'action__path': entry,
                 })
+            if context.show_parent:
+                candidates.append({'word':'../','is_directory':True, 'action__path':path.resolve().parent,})
```
Comment (Shougo) [inline rplugin/python3/defx/source/file/__init__.py:64]: `candidates.append({'word' :'../', 'is_directory': True, 'action__path': path.resolve().parent})`

File: doc/defx.txt (hunk 2 above)
Comment (gamark) [reply doc/defx.txt]: > `Show parent "../" directory in the buffer` is better for me.

get.

File: rplugin/python3/defx/source/file/__init__.py (hunk 4 above)
Comment (Shougo) [reply rplugin/python3/defx/source/file/__init__.py:64]: And please fix the lint.

File: rplugin/python3/defx/context.py [hunk 5]
```diff
@@ -37,6 +37,7 @@ class Context(typing.NamedTuple):
     search_recursive: str = ''
     session_file: str = ''
     show_ignored_files: bool = False
+    show_parent:bool = False
```
Comment (Shougo) [inline rplugin/python3/defx/context.py:40]: Please add the space like above.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**